pkg/extractor/index.go

```diff
@@ -0,0 +1,36 @@
+package extractor
+
+import (
+	"context"
+	"sort"
+
+	"github.com/mt-sre/addon-metadata-operator/pkg/types"
+	"github.com/operator-framework/operator-registry/alpha/action"
+)
+
+type indexExtractor struct {
+	types.IndexExtractorCache
+}
+
+func NewIndexExtractor(cache types.IndexExtractorCache) types.IndexExtractor {
+	return indexExtractor{cache}
+}
+
+// ListBundles - return a list of all bundle images, sorted
+func (i indexExtractor) ListBundles(indexImage string, packageName string) ([]string, error) {
```

Comment:
@ashishmax31 @yashvardhan-kukreja  Should this be a pointer receiver? As it modifies the `indexExtractor` object, populating it's cache?

Reply:
I went ahead and made it a pointer receiver, as it is modifying the state of the internal cache structure.

Reply:
@sugarraysam I don't think its required since `map` is a reference object by default.

Reply:
Digging into go FAQ, even for a `map`, which is a reference type object, its best practice to make the method receiver a pointer if it modifies internal state: https://go.dev/doc/faq#methods_on_values_or_pointers

> First, and most important, does the method need to modify the receiver? If it does, the receiver must be a pointer.